CMakeLists.txt

```diff
@@ -287,6 +288,8 @@ set_target_properties(rocksdb${ARTIFACT_SUFFIX} PROPERTIES COMPILE_FLAGS "-DROCK
 add_dependencies(rocksdb${ARTIFACT_SUFFIX} GenerateBuildVersion)
 target_link_libraries(rocksdb${ARTIFACT_SUFFIX} ${LIBS})
 
+include(${CMAKE_CURRENT_SOURCE_DIR}/java/jni.inc)
```

Comment:
@koldat  The main point here is not to make the main file cluttered. So a separate file and a separate folder is  a good idea. Since many do not build java here is additional ideas.

1. Include this file optionally based on the command line option OR
1.  Since you say it is a separate project you can instead of a include file write it in a separate CMakeLists.txt in that subfolder and conditionally add it as a separate project based on the command line. Like here `rocksdb/third-party/gtest-1.7.0/fused-src/gtest using add_subdirectory